runners/java-fn-execution/src/test/java/org/apache/beam/runners/fnexecution/control/SingleEnvironmentInstanceJobBundleFactoryTest.java

```diff
@@ -76,7 +76,8 @@ public void setup() throws Exception {
 
     InProcessServerFactory serverFactory = InProcessServerFactory.create();
     dataServer =
-        GrpcFnServer.allocatePortAndCreateFor(GrpcDataService.create(executor), serverFactory);
+        GrpcFnServer.allocatePortAndCreateFor(
+            GrpcDataService.create(executor, obs -> obs), serverFactory);
```

Comment:
Please swap where you did `obs -> obs` with the `direct` StreamObserverFactory. It is likely to reduce flaky tests across the board.

Reply:
Done.